plutus-exe/src/Main.hs

```diff
@@ -107,9 +156,58 @@ runEval (EvalOptions inp mode) = do
             T.putStrLn $ PLC.prettyPlcDefText v
             exitSuccess
 
+data TypeExample = TypeExample (PLC.Kind ()) (PLC.Type PLC.TyName ())
+data TermExample = TermExample (PLC.Type PLC.TyName ()) (PLC.Term PLC.TyName PLC.Name ())
+data SomeExample = SomeTypeExample TypeExample | SomeTermExample TermExample
+
+prettySignature :: ExampleName -> SomeExample -> Doc ann
+prettySignature name (SomeTypeExample (TypeExample kind _)) =
+    pretty name <+> "::" <+> PLC.prettyPlcDef kind
+prettySignature name (SomeTermExample (TermExample ty _)) =
+    pretty name <+> ":"  <+> PLC.prettyPlcDef ty
+
+prettyExampleBody :: SomeExample -> Doc ann
+prettyExampleBody (SomeTypeExample (TypeExample _ ty))   = PLC.prettyPlcDef ty
+prettyExampleBody (SomeTermExample (TermExample _ term)) =
+    PLC.prettyPlcDef $ PLC.Program () (PLC.defaultVersion ()) term
+
+prettyExample :: ExampleOutputMode -> ExampleName -> SomeExample -> Doc ann
+prettyExample ExampleOutputLegalPlc     _    example = prettyExampleBody example
+prettyExample ExampleOutputDeclarations name example =
+    prettySignature name example <> hardline <> pretty name <+> "=" <+> prettyExampleBody example
+
+avalaibleExamples :: [(ExampleName, SomeExample)]
```

Comment:
Because I'm manually specifying kinds and types. I shouldn't do this at all, but I do not like the interface to our type/kind checker which is either "here is `rename`, `annotate` and `typecheck` for you, compose them in the right order" or "give me a `ByteString`" or "well, you can give me a `Prrogram`, but I'll check whether types are normalized", and because I want to give only simple examples for now. `foldStdLib` is a sensible thing, perhaps.

I plan to clean up the type checker API once I have some free time.

Reply:
Okay. Note that you can now pass the `TypeConfig` to the typechecker pipeline and it will only check for normalized types if it should do.